### PR TITLE
[176] Adapt SiriusDiagramActionBarContributor to avoid memory leak

### PR DESCRIPTION
**Pull request by lredor**
This PR follows PR #175.

Create a specific ActionBarContributor for each specific implementation of SiriusDiagramEditor to avoid a duplication of ActionRegistry: one for "SiriusDiagramEditor.class" and one for real implementation ("DDiagramEditorImpl.class" or "EcoreEntitiesReadOnlyEditor.class"). Indeed, only one of the ActionRegistry is correctly cleaned at the closing of the editor.

Bug: https://github.com/eclipse-sirius/sirius-desktop/issues/176

### CODE REVIEW

**Comment by lredor** [comment]
> continuous-integration/jenkins/pr-head — This commit cannot be built [Details](https://ci.eclipse.org/sirius/job/sirius-pr-check/job/PR-177/2/display/redirect)

Two remarks:
* The 3 failures are OK locally. So the PR seems not in cause. I launched another job to check if the problems are always here.
* The SWTBot jobs seem longer than usual:
    * Sample for PR #175 
image
    * Sample for current PR
 
image

To check before merging, this is maybe a sign of performance problem caused by this PR. Wait and see with another PR in progress (PR #181 with [this job](https://ci.eclipse.org/sirius/job/sirius-pr-check/view/change-requests/job/PR-181/)).

**Review by pcdavid — APPROVED** [review]

**Comment by pcdavid** [comment]
There are 2 test failures on the CI but they are OK when I run them locally, so I guess we can merge.

**Comment by lredor** [comment]
> There are 2 test failures on the CI but they are OK when I run them locally, so I guess we can merge.

As said in a [previous comment](https://github.com/eclipse-sirius/sirius-desktop/pull/177#issuecomment-1833444655), the memory leak fix maybe implies performance problem (the tests execution duration seems longer for this PR). So I'm not sure it is to be merged before checking this aspect.

**Comment by lredor** [comment]
> To check before merging, this is maybe a sign of performance problem caused by this PR. Wait and see with another PR in progress (PR https://github.com/eclipse-sirius/sirius-desktop/pull/181 with [this job](https://ci.eclipse.org/sirius/job/sirius-pr-check/view/change-requests/job/PR-181/)).

I've analyzed the log and indeed several tests are longer than usual. But locally, when I launched this kind of tests, the time is similar with or without this memory leak fix. Whereas on the server the times observed are sometimes ten times longer. So I have no real explanation. The commit has been rebased. I will see the result with the [current job](https://ci.eclipse.org/sirius/job/sirius-pr-check/job/PR-177/9/)

**Comment by lredor** [comment]
I currently convert to DRAFT as this memory leak fix needs analysis about time execution. As it is not my main task, I don't know when I really do that. I have added a commit to reduce the suite to compare time on less tests (comparison to done between this PR and PR #188).

**Comment by lredor** [comment]
The last "force push" is just a rebase on master branch (7.4.2). No additional review is needed as there is no change. I'll merge this PR after a "green" job of CI.

**Comment by lredor** [comment]
The last "force push" is just a rebase on master branch (7.4.3). No additional review is needed as there is no change. I'll merge this PR after a "green" job of CI.

**Pull request merged**
